auto_dev/data/repo/templates/autonomy/scripts/run_single_agent.sh

```diff
@@ -0,0 +1,152 @@
+    echo "  --force            : Remove existing build directory if it exists"
+    echo "  --debug            : Run the agent in DEBUG mode"
+    exit 1
+fi
+
+# Initialize variables
+FORCE=false
+DEBUG=false
+AGENT_IDENTIFIER=""
+
+# Parse arguments
+while [[ "$#" -gt 0 ]]; do
+    case "$1" in
+        --force)
+            FORCE=true
+            ;;
+        --debug)
+            DEBUG=true
+            ;;
+        -*)
+            echo "Unknown option: $1"
+            echo "Usage: $0 <author/agent_name> [--force] [--debug]"
+            exit 1
+            ;;
+        *)
+            if [[ -z "$AGENT_IDENTIFIER" ]]; then
+                AGENT_IDENTIFIER="$1"
+            else
+                echo "Multiple agent identifiers provided."
+                echo "Usage: $0 <author/agent_name> [--force] [--debug]"
+                exit 1
+            fi
+            ;;
+    esac
+    shift
+done
+
+agent_author=$(echo "$AGENT_IDENTIFIER" | cut -d'/' -f1)
+agent_name=$(echo "$AGENT_IDENTIFIER" | cut -d'/' -f2)
+
+
+echo "   Agent author: $agent_author"
+echo "   Agent name:   $agent_name"
+
+CWD=$(pwd)
+BUILD_DIR="build"
+AGENT_BUILD_DIR="$BUILD_DIR/$agent_author/$agent_name"
+
+# Handle existing build directory
+if [[ -d "$AGENT_BUILD_DIR" ]]; then
+    if [[ "$FORCE" == true ]]; then
+        echo "Removing existing build directory '$AGENT_BUILD_DIR' due to --force."
+        rm -rf "$AGENT_BUILD_DIR"
+    else
+        echo "Error: Build directory '$AGENT_BUILD_DIR' already exists. Use --force to overwrite."
+        exit 1
+    fi
+fi
+
+# Ensure the build author directory exists
+mkdir -p "$BUILD_DIR/$agent_author"
+
+# Fetch the agent from the local package registry
+echo "Fetching agent '$AGENT_IDENTIFIER' from the local package registry..."
+if ! aea -s fetch "$AGENT_IDENTIFIER" --local > /dev/null; then
+    echo "Error: Failed to fetch agent '$AGENT_IDENTIFIER'."
+    exit 1
+fi
+
+# Move the fetched agent to the build directory
+echo "Moving agent to '$AGENT_BUILD_DIR'."
+mv "$agent_name" "$BUILD_DIR/$agent_author/"
+
+# Navigate to the agent's build directory
+cd "$AGENT_BUILD_DIR"
+
+# create and add a new ethereum key
+echo "Setting up Ethereum key..."
+if [[ ! -f "$CWD/ethereum_private_key.txt" ]]; then
+    if ! aea -s generate-key ethereum || ! aea -s add-key ethereum; then
+        echo "Error: Failed to generate or add Ethereum key."
+        exit 1
+    fi
+else
+    cp "$CWD/ethereum_private_key.txt" "./ethereum_private_key.txt"
+    if ! aea -s add-key ethereum; then
+        echo "Error: Failed to add Ethereum key."
+        exit 1
+    fi
+fi
+
+# install any agent deps
+echo "Installing agent dependencies..."
+if ! aea -s install; then
+    echo "Error: Failed to install agent dependencies."
+    exit 1
+fi
+
+# issue certificates for agent peer-to-peer communications
+if [[ ! -d "$CWD/certs" ]]; then
+    echo "Issuing certificates for agent peer-to-peer communications..."
+    if ! aea -s issue-certificates; then
```

Comment:
Not all agents need certs

Reply:
mb flag to disable?

Reply:
gosh no, assumes way to o much on the user :) 
There is a key to check for iirc

Reply:
keys automatically generated here https://github.com/8ball030/auto_dev/blob/50fe35428d35c80dcb2660373aea45c3ed5bd2bc/auto_dev/data/repo/templates/autonomy/scripts/run_single_agent.sh#L83-L96